### ISSUE TITLE
Custom class names randomly (and permanently) stop autocompleting

### ISSUE THREAD

**Issue by LO-0116** [godotengine/godot#78926]
### Godot version

v4.0.2.stable.official [7a0977ce2]

### System information

Windows 10, compatibility renderer

### Issue description

Sometimes, a custom class declared by `class_name` will stop updating with changes to the class's file.

For example, if you have a class `Foo`, and then add a variable to it like `fooVar`, and then try to index that variable from other contexts, such as from other scripts extended Foo or other scripts that type hints Foo, the new `fooVar` will not appear. Same goes for added functions.

Once this failure to update occurs once, it will always fail for all future changes to this class (other classes in the same project are not affected). Reloading the project will update changes to the class, but only once. Each added variable or function requires a reload to appear elsewhere.

**I cannot reproduce this with minimal projects, only during serious production** Which also makes it impossible for me to share the code due to NDA.

Something observations:

1. this bug only appears when a cyclic reference between two custom classes has occurred. Removing cyclic reference does not fix it.
2. moving the offending files and scenes to a different (nearly empty) project will solve the problem

### Steps to reproduce

N/A: bug only encountered during serious production. Minimal projects cannot reproduce bug.

### Minimal reproduction project

N/A

**Comment by dalexeev** [comment]
* Duplicate of #76380.
